backend/src/test/java/com/huellapositiva/api/ProposalControllerShould.java

```diff
@@ -89,6 +93,90 @@ void beforeEach() {
+        JpaProposal publishedProp= testData.registerESALAndProposal(PUBLISHED);
+        publishedProp.setClosingProposalDate(new SimpleDateFormat("dd-MM-yyyy").parse("20-12-2020"));
+
+        // WHEN
+        proposalService.changeStatusToFinished(publishedProp.getId());
+
+        // THEN
+        assertThat(jpaProposalRepository.findByNaturalId(publishedProp.getId()).get().getStatus().getName()).isEqualTo("finished");
+    }
+
+    @Test
+    void return_409_when_inadequate_criteria_to_change_proposal_status_to_finished() throws Exception{
+        // GIVEN
+        JpaProposal publishedProp= testData.registerESALAndProposal(CANCELLED);
+        publishedProp.setClosingProposalDate(new SimpleDateFormat("dd-MM-yyyy").parse("20-12-2020"));
+
+        JwtResponseDto jwtResponseDto = loginAndGetJwtTokens(mvc, DEFAULT_ESAL_CONTACT_PERSON_EMAIL, DEFAULT_PASSWORD);
+
+
+        // WHEN + THEN
+        mvc.perform(post(FETCH_PROPOSAL_URI + publishedProp.getId() + "/status/toFinished")
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + jwtResponseDto.getAccessToken())
+                .contentType(MediaType.APPLICATION_JSON)
+                .with(csrf())
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isConflict())
+                .andExpect(mvcResult -> assertThat(jpaProposalRepository.findByNaturalId(publishedProp.getId()).get().getStatus().getName()).isEqualTo("cancelled"));
+
+    }
+
+    @Test
+    void return_200_when_adequate_criteria_to_change_proposal_status_to_finished() throws Exception{
+        // GIVEN
+        JpaProposal publishedProp= testData.registerESALAndProposal(PUBLISHED);
+        publishedProp.setClosingProposalDate(new SimpleDateFormat("dd-MM-yyyy").parse("20-12-2020"));
+
+        JwtResponseDto jwtResponseDto = loginAndGetJwtTokens(mvc, DEFAULT_ESAL_CONTACT_PERSON_EMAIL, DEFAULT_PASSWORD);
+
+
+        // WHEN + THEN
+        mvc.perform(post(FETCH_PROPOSAL_URI + publishedProp.getId() + "/status/toFinished")
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + jwtResponseDto.getAccessToken())
+                .contentType(MediaType.APPLICATION_JSON)
+                .with(csrf())
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(mvcResult -> assertThat(jpaProposalRepository.findByNaturalId(publishedProp.getId()).get().getStatus().getName()).isEqualTo("finished"));
+
+    }
+
+    @Test
+    void trigger_bulk_status_change_to_finished() throws Exception {
+
+        // GIVEN
+        JpaProposal finishedProp= testData.registerESALAndProposal(FINISHED);
+        finishedProp.setClosingProposalDate(new SimpleDateFormat("dd-MM-yyyy").parse("20-12-2020"));
+
+        testData.resetData();
+
+        JpaProposal publishedProp= testData.registerESALAndProposal(PUBLISHED);
+        publishedProp.setClosingProposalDate(new SimpleDateFormat("dd-MM-yyyy").parse("20-12-2099"));
```

Comment:
This test will break in 70 years. You should never hardcode dates in your tests. The test can only have one reason to fail, they need to be bulletproof.

`publishedProp.setClosingProposalDate(Date.from(Instant.now().minus(3, MONTH)));`